aspnetcore/mvc/views/working-with-forms.md

```diff
@@ -184,6 +184,8 @@ Generates the following:
 
 With collection properties, `asp-for="CollectionProperty[23].Member"` generates the same name as `asp-for="CollectionProperty[i].Member"` when `i` has the value `23`.
 
+The calculated value of a `ModelExpression` is the first non-null result found in the ModelState entry with full name, the ViewData entry with full name, or the expression evaluated against Model, in that order. 
```

Comment:
This is not correct in almost all cases. Instead, look at [`TextBoxFor(...)`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering.ihtmlhelper-1.textareafor?view=aspnetcore-2.0) for one fairly-general case.

Also need to be a bit more generic. Different tag helpers and different `<input/>` `type`s handle the expressions differently. For example, the `value` of a generated `<input type="checkbox" .../>` is always `true` and the expression is instead evaluated to determine `checked` (see [`CheckBoxFor(...)`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering.ihtmlhelper-1.checkboxfor?view=aspnetcore-2.0)). `<input type="password" .../>` ignores the expression except to name the generated `<input/>` (see [`PasswordFor(...)`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering.ihtmlhelper-1.passwordfor?view=aspnetcore-2.0)). And, yes, I could go on because there are few all-encompassing rules here.